File: property-service/src/main/java/fr/urbanmarkets/propertyservice/model/Property.java
package fr.urbanmarkets.propertyservice.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import java.util.HashSet;
import java.util.Set;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "property")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Property {

  @Id
  @Column(name = "propertyId", nullable = false)
  private String propertyId;

  @Column(name = "surface")
  private Integer surface;

  @Column(name = "bedroom_count")
  private Integer bedroomCount;

  @OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
  @JoinColumn(name = "addressId")
  @JsonManagedReference
  private Address address;

  @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, orphanRemoval = true)
  @JoinColumn(name="propertyId")
  private Set<Price> prices = new HashSet<>();

}

File: gateway-service/src/main/angular/ui/src/app/property.ts
import {Address} from "./address";
import {Price} from "./price";

export class Property {
  propertyId: String;
  surface: Number;
  bedroomCount: Number;
  address: Address;
  prices: Array<Price>;
  constructor(propertyId: String, surface: Number, bedroomCount: Number, address: Address,
              prices: Array<Price>){
    this.propertyId = propertyId;
    this.surface = surface;
    this.bedroomCount = bedroomCount;
    this.address = address;
    this.prices = prices;
  }
}
